.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,220 @@
+name: e2e
+
+on:
+  push:
+    branches:
+      - 'master'
+    tags:
+      - 'v*'
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+
+env:
+  GO_VERSION: "1.21.4"
+  K8S_VERSION: "v1.27.3"
+  KIND_CLUSTER_NAME: "kind"
+
+jobs:
+  build:
+    name: build
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.GO_VERSION }}
+      id: go
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Build
+      run: |
+        mkdir _output
+
+        docker build -t gcr.io/k8s-staging-kas-network-proxy/proxy-agent:master -f artifacts/images/agent-build.Dockerfile .
+        docker save gcr.io/k8s-staging-kas-network-proxy/proxy-agent:master > _output/konnectivity-agent.tar
+        docker build -t gcr.io/k8s-staging-kas-network-proxy/proxy-server:master -f artifacts/images/server-build.Dockerfile .
+        docker save gcr.io/k8s-staging-kas-network-proxy/proxy-server:master >  > _output/konnectivity-server.tar
+
+    - uses: actions/upload-artifact@v2
+      with:
+        name: konnectivity-server
+        path: _output/konnectivity-server.tar
+
+    - uses: actions/upload-artifact@v2
+      with:
+        name: konnectivity-agent
+        path: _output/konnectivity-agent.tar
+  e2e:
+    name: e2e
+    runs-on: ubuntu-20.04
+    timeout-minutes: 100
+    needs:
+      - build
+    strategy:
+      fail-fast: false
+      matrix:
+        ipFamily: ["ipv4", "ipv6", "dual"]
```

Comment:
in the ipv6 case, is the `httpd` pod below only given an IPv6 IP address? It might be nice to print it out for inspection.

Reply:
Yes, that is guaranteed with kind ci, we can give that for granted